Expandable console support

jacob314 opened the pull request.

jacob314 reviewed (commented):
We should rewrite how we are testing the debugger to use Flutter integration tests similar to what has been done for the inspector tests. That will let us use golden image tests to verify the UI state.

jacob314 commented:
Fyi @grouma, @annagrin

kenzieschmoll commented:
can you add a screenshot or gif to your PR description? thanks

grouma commented on packages/devtools_app/lib/src/tree.dart:
```diff
@@ -13,10 +13,27 @@ class TreeView<T extends TreeNode<T>> extends StatefulWidget {
+  /// scroll view needs to be recomputed whenever the scroll position changes.
+  ///
+  /// Defaults to false.
+  final bool shrinkWrap;
```
Is this a copied comment? It doesn't seem appropriate here?

jacob314 replied:
Yeah this is the boilerplate Flutter comment on what shrinkWrap does. I have now swapped to the TLDR of when it is needed.

jacob314 commented:
Here is a screenshot.
<img width="1665" alt="Screen Shot 2021-02-02 at 8 51 24 AM" src="https://user-images.githubusercontent.com/1226812/106643194-36668c00-653e-11eb-97e0-cd95650dafbf.png">

jacob314 commented:
One note is the console has a mix of fonts. We could make everything monospace if we want.

jacob314 commented:
I think we should make all expandable variables in the debugger monospace font for consistency. Tweaking that.

jacob314 commented:
Please review the follow up changes since the last review. I needed to make a small refactor to resolve fighting between SelectableText widgets and InkWell widgets. Also fixed some tests that were broken and made JSON dumps fully selectable.
